pkg/service/admission/externalip_admission.go

```diff
@@ -103,8 +126,58 @@ func (r *externalIPRanger) Admit(a kadmission.Attributes) error {
 			}
 		}
 	}
+
+	// Attempt to allocate the specified external ips
+	// TODO cleanup newly allocated ips on failure
+	if len(errs) == 0 {
+		for i, ip := range svc.Spec.ExternalIPs {
+			err := r.allocateExternalIP(ip, svc.Namespace, svc.Name)
+			if err != nil {
+				if err == externalIPInUse {
+					errs = append(errs, field.Duplicate(field.NewPath("spec", "externalIPs").Index(i), err.Error()))
+				} else {
+					errs = append(errs, field.InternalError(field.NewPath("spec", "externalIPs").Index(i), err))
+				}
+				// Skip allocating ips after an allocation failure
+				break
+			}
+		}
+	}
+
 	if len(errs) > 0 {
 		return apierrs.NewInvalid(a.GetKind().GroupKind(), a.GetName(), errs)
 	}
 	return nil
 }
+
+// allocateExternalIP will attempt to allocate the given ip address
+// for the service identified by serviceNamespace and serviceName.
+// The return value will be nil if allocation was successful, or an
+// error otherwise.
+func (r *externalIPRanger) allocateExternalIP(ip, serviceNamespace, serviceName string) error {
```

Comment:
We really don't want to add more admission controllers, especially when this can be done in a regular controller.  Basically, you want to do something like the Security Allocator (controller that allocates things onto the namespace).

Once I know more about how you intend these to be consumed by end users, I'll be able to offer more advice.